src/App.tsx

```diff
@@ -1,110 +1,45 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.css';
-import { getNumbers } from './utils';
+import { getNumbers, getPartItems } from './utils';
+import { Pagination } from './components/Pagination';
+import { DropDown } from './components/DropDown/DropDown';
+import { ElementsOnPage } from './components/ElementsOnPage/ElementsOnPage';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(5);
+  const [page, setPage] = useState(1);
+  const [elements, setElements] = useState<string[]>([]);
+
+  const startRange = elements[0] ? elements[0].split(' ')[1] : 1;
+  const endRange = elements[elements.length - 1]
+    ? elements[elements.length - 1].split(' ')[1]
+    : 5;
+
+  useEffect(() => {
+    const updatedElements = getPartItems(items, page, perPage);
+
+    setElements(updatedElements);
+  }, [perPage, setElements, page]);
```

Comment:
```suggestion

    const updatedElements = getPartItems(items, page, perPage);
```
and just reuse this variable where you need it